tests/test_snapshot_scheduler.cpp

```diff
@@ -83,25 +83,30 @@ BOOST_AUTO_TEST_CASE(snapshot_scheduler_test) {
             if (!pp->get_snapshot_requests().snapshot_requests.empty()) {
                const auto& snapshot_requests = pp->get_snapshot_requests().snapshot_requests;
 
-               auto validate_snapshot_request = [&](uint32_t sid, uint32_t block_num, uint32_t spacing = 0) {                  
+               auto validate_snapshot_request = [&](uint32_t sid, uint32_t block_num, uint32_t spacing = 0, bool fuzzy_start = false) {                  
                   auto it = find_if(snapshot_requests.begin(), snapshot_requests.end(), [sid](const snapshot_scheduler::snapshot_schedule_information& obj) {return obj.snapshot_request_id == sid;});
                   if (it != snapshot_requests.end()) {
                      auto& pending = it->pending_snapshots;
                      if (pending.size()==1) {
+                        // pending snapshot block number
                         auto pbn = pending.begin()->head_block_num;
-                        pbn = spacing ?  (spacing + (pbn%spacing)) : pbn;
-                        // if snapshot scheduled with empty start_block_num depending on the timing
-                        // it can be scheduled either for block_num or block_num+1
-                        BOOST_CHECK(block_num==pbn || ((block_num+1)==pbn));                       
+
+                        // first pending snapshot
+                        auto ps_start = spacing ?  (spacing + (pbn%spacing)) : pbn;
```

Comment:
As `spacing` is used in modular division `pbn%spacing`, it is more readable 
```
auto ps_start = (spacing != 0) ? (spacing + (pbn%spacing)) : pbn;
```

Reply:
Done, thank you!